aten/src/ATen/native/sparse/SparseTensorMath.cpp

```diff
@@ -501,15 +574,15 @@ SparseTensor& mul_out_sparse_cpu(SparseTensor& r, const Tensor& t_, const Tensor
   if (t_values.dim() > 1) {
     while (t_i < t_nnz && s_i < s_nnz) {
       if (!index_preamble()) continue;
-      r_values.select(0, r_i).addcmul_(t_values.select(0, t_i), s_values.select(0, s_i));
+      r_buffer.select(0, r_i).addcmul_(t_values.select(0, t_i), s_values.select(0, s_i));
```

Comment:
Is type promotion implemented on `addcmul_`? If it is then we don't need to cast `t._values()` and `src._values()` to the promoted dtype.

Reply:
not implemented